dz3.py

```diff
@@ -0,0 +1,182 @@
+#!python3
+import copy
+
+
+class Url(object):
+
+    def __init__(self, scheme='', authority='', path='', query='', fragment=''):
+        self.scheme = scheme
+        self.authority = authority
+        self.path = path
+        self.query = query
+        self.fragment = fragment
+
+    def __str__(self):
+        if isinstance(self.path, str):
```

Comment:
return f"{self.scheme}://{self.authority}/{'/'.join(p in self.path)}/{'&.join(...)"